src/bluesky/consolidators.py

```diff
@@ -271,21 +272,22 @@ def adapter_parameters(self) -> Dict:
         return {"dataset": self._sres_parameters["dataset"].strip("/").split("/"), "swmr": self.swmr}
 
 
-class TIFFConsolidator(ConsolidatorBase):
-    supported_mimetypes = {"multipart/related;type=image/tiff"}
-
-    def __init__(self, stream_resource: StreamResource, descriptor: EventDescriptor):
+class MultipartRelatedConsolidator(ConsolidatorBase):
+    def __init__(self, permitted_extensions: str, stream_resource: StreamResource, descriptor: EventDescriptor):
         super().__init__(stream_resource, descriptor)
+        self.permitted_extensions: str = permitted_extensions
         self.data_uris: List[str] = []
 
     def get_datum_uri(self, indx: int):
-        """Return a full uri for a datum (an individual TIFF file) based on its index in the sequence.
+        """Return a full uri for a datum (an individual image file) based on its index in the sequence.
 
         This relies on the `template` parameter passed in the StreamResource, which is a string either in the "new"
         Python formatting style that can be evaluated to a file name using the `.format(indx)` method given an
-        integer index, e.g. "{:05d}.tif".
+        integer index, e.g. "{:05d}.ext".
         """
-        return self.uri + self._sres_parameters["template"].format(indx)
+        filename_template = self._sres_parameters["template"]
```

Comment:
I was actually going to move this line to `__init__`, do some extra validation for back-compatibility there, and then just use `self.template` in get_datum_uri. I think you changes here would fall nicely in that pattern.